tsl/src/compression/create.c

```diff
@@ -1067,6 +1067,14 @@ tsl_process_compress_table_add_column(Hypertable *ht, ColumnDef *orig_def)
 
 	coloid = LookupTypeNameOid(NULL, orig_typname, false);
 	compresscolinfo_init_singlecolumn(&compress_cols, colname, coloid);
+
+	FormData_hypertable_compression *ht_comp =
```

Comment:
I would move the newly added lines to line 1067.
So the check would be the first thing in the function (before any initialization etc) .